<a href="https://colab.research.google.com/github/TheJoys2019/DS-Unit-1-Sprint-2-Data-Wrangling/blob/master/Artin%20Sinani%20-%20DS_Unit_1_Sprint_Challenge_2_Data_Wrangling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Data Science Unit 1 Sprint Challenge 2

## Data Wrangling

In this Sprint Challenge you will use data from [Gapminder](https://www.gapminder.org/about-gapminder/), a Swedish non-profit co-founded by Hans Rosling. "Gapminder produces free teaching resources making the world understandable based on reliable statistics."
- [Cell phones (total), by country and year](https://raw.githubusercontent.com/open-numbers/ddf--gapminder--systema_globalis/master/ddf--datapoints--cell_phones_total--by--geo--time.csv)
- [Population (total), by country and year](https://raw.githubusercontent.com/open-numbers/ddf--gapminder--systema_globalis/master/ddf--datapoints--population_total--by--geo--time.csv)
- [Geo country codes](https://github.com/open-numbers/ddf--gapminder--systema_globalis/blob/master/ddf--entities--geo--country.csv)

These two links have everything you need to successfully complete the Sprint Challenge!
- [Pandas documentation: Working with Text Data](https://pandas.pydata.org/pandas-docs/stable/text.html]) (one question)
- [Pandas Cheat Sheet](https://github.com/pandas-dev/pandas/blob/master/doc/cheatsheet/Pandas_Cheat_Sheet.pdf) (everything else)

## Part 0. Load data

You don't need to add or change anything here. Just run this cell and it loads the data for you, into three dataframes.

In [0]:
import pandas as pd

cell_phones = pd.read_csv('https://raw.githubusercontent.com/open-numbers/ddf--gapminder--systema_globalis/master/ddf--datapoints--cell_phones_total--by--geo--time.csv')

population = pd.read_csv('https://raw.githubusercontent.com/open-numbers/ddf--gapminder--systema_globalis/master/ddf--datapoints--population_total--by--geo--time.csv')

geo_country_codes = (pd.read_csv('https://raw.githubusercontent.com/open-numbers/ddf--gapminder--systema_globalis/master/ddf--entities--geo--country.csv')
                       .rename(columns={'country': 'geo', 'name': 'country'}))

## Part 1. Join data

First, join the `cell_phones` and `population` dataframes (with an inner join on `geo` and `time`).

The resulting dataframe's shape should be: (8590, 4)

In [3]:
cell_phones.columns

Index(['geo', 'time', 'cell_phones_total'], dtype='object')

In [4]:
cell_phones.shape

(9215, 3)

In [5]:
cell_phones.head()

geo  time  cell_phones_total
0  abw  1960                0.0
1  abw  1965                0.0
2  abw  1970                0.0
3  abw  1975                0.0
4  abw  1976                0.0

In [0]:
cp1 = pd.DataFrame(cell_phones, columns=['geo', 'time'])

In [0]:
pop1 = pd.DataFrame(population, columns=['geo','time'])

In [8]:
population.head()

geo  time  population_total
0  afg  1800           3280000
1  afg  1801           3280000
2  afg  1802           3280000
3  afg  1803           3280000
4  afg  1804           3280000

In [9]:
population.columns

Index(['geo', 'time', 'population_total'], dtype='object')

In [0]:
cp = pd.merge(cell_phones, population)

In [11]:
cp.shape

(8590, 4)

In [12]:
cp.head()

geo  time  cell_phones_total  population_total
0  afg  1960                0.0           8996351
1  afg  1965                0.0           9938414
2  afg  1970                0.0          11126123
3  afg  1975                0.0          12590286
4  afg  1976                0.0          12840299

In [0]:
cell_pop = pd.merge(cell_phones[['geo','time']], population[['geo','time']])

In [14]:
cell_pop.shape

(8590, 2)

Then, select the `geo` and `country` columns from the `geo_country_codes` dataframe, and join with your population and cell phone data.

The resulting dataframe's shape should be: (8590, 5)

In [15]:
geo_country_codes.head()

geo alt_5                    alternative_1 alternative_2 alternative_3  \
0  abkh   NaN                              NaN           NaN           NaN   
1   abw   NaN                              NaN           NaN           NaN   
2   afg   NaN  Islamic Republic of Afghanistan           NaN           NaN   
3   ago   NaN                              NaN           NaN           NaN   
4   aia   NaN                              NaN           NaN           NaN   

  alternative_4_cdiac arb1 arb2 arb3 arb4         ...           latitude  \
0                 NaN  NaN  NaN  NaN  NaN         ...                NaN   
1               Aruba  NaN  NaN  NaN  NaN         ...           12.50000   
2         Afghanistan  NaN  NaN  NaN  NaN         ...           33.00000   
3              Angola  NaN  NaN  NaN  NaN         ...          -12.50000   
4                 NaN  NaN  NaN  NaN  NaN         ...           18.21667   

  longitude main_religion_2008      country        pandg un_state  \
0       NaN                NaN     Abkhazia          NaN    False   
1 -69.96667          christian        Aruba          NaN    False   
2  66.00000             muslim  Afghanistan  AFGHANISTAN     True   
3  18.50000          christian       Angola       ANGOLA     True   
4 -63.05000          christian     Anguilla          NaN    False   

  unicode_region_subtag  upper_case_name world_4region        world_6region  
0                   NaN              NaN        europe  europe_central_asia  
1                    AW            ARUBA      americas              america  
2                    AF      AFGHANISTAN          asia           south_asia  
3                    AO           ANGOLA        africa   sub_saharan_africa  
4                    AI         ANGUILLA      americas              america  

[5 rows x 33 columns]

In [0]:
df1 = pd.DataFrame(geo_country_codes, columns=['geo', 'country'])

In [17]:
df1.head()

geo      country
0  abkh     Abkhazia
1   abw        Aruba
2   afg  Afghanistan
3   ago       Angola
4   aia     Anguilla

In [0]:
df1_cp = pd.merge(df1,cp)

In [19]:
df1_cp.shape

(8590, 5)

## Part 2. Make features

Calculate the number of cell phones per person, and add this column onto your dataframe.

(You've calculated correctly if you get 1.220 cell phones per person in the United States in 2017.)

In [0]:
df1_cp['avg_cell_pop'] = df1_cp['cell_phones_total'] / df1_cp['population_total']

In [21]:
df1_cp['avg_cell_pop'].tail()

8585    0.905587
8586    0.765566
8587    0.808585
8588    0.797439
8589    0.852522
Name: avg_cell_pop, dtype: float64

In [22]:
df1_cp.loc[df1_cp['country']== 'United States']

geo        country  time  cell_phones_total  population_total  \
8092  usa  United States  1960                0.0         186808228   
8093  usa  United States  1965                0.0         199815540   
8094  usa  United States  1970                0.0         209588150   
8095  usa  United States  1975                0.0         219205296   
8096  usa  United States  1976                0.0         221239215   
8097  usa  United States  1977                0.0         223324042   
8098  usa  United States  1978                0.0         225449657   
8099  usa  United States  1979                0.0         227599878   
8100  usa  United States  1980                0.0         229763052   
8101  usa  United States  1984            91600.0         238573861   
8102  usa  United States  1985           340213.0         240824120   
8103  usa  United States  1986           681825.0         243098935   
8104  usa  United States  1987          1230855.0         245402864   
8105  usa  United States  1988          2069441.0         247739582   
8106  usa  United States  1989          3508944.0         250113187   
8107  usa  United States  1990          5283055.0         252529950   
8108  usa  United States  1991          7557148.0         254974819   
8109  usa  United States  1992         11032753.0         257454273   
8110  usa  United States  1993         16009461.0         260020186   
8111  usa  United States  1994         24134421.0         262741566   
8112  usa  United States  1995         33785661.0         265658849   
8113  usa  United States  1996         44042992.0         268803424   
8114  usa  United States  1997         55312293.0         272136551   
8115  usa  United States  1998         69209321.0         275542603   
8116  usa  United States  1999         86047003.0         278862277   
8117  usa  United States  2000        109478031.0         281982778   
8118  usa  United States  2001        128500000.0         284852391   
8119  usa  United States  2002        141800000.0         287506847   
8120  usa  United States  2003        160637000.0         290027624   
8121  usa  United States  2004        184819000.0         292539324   
8122  usa  United States  2005        203700000.0         295129501   
8123  usa  United States  2006        229600000.0         297827356   
8124  usa  United States  2007        249300000.0         300595175   
8125  usa  United States  2008        261300000.0         303374067   
8126  usa  United States  2009        274283000.0         306076362   
8127  usa  United States  2010        285118000.0         308641391   
8128  usa  United States  2011        297404000.0         311051373   
8129  usa  United States  2012        304838000.0         313335423   
8130  usa  United States  2013        310698000.0         315536676   
8131  usa  United States  2014        355500000.0         317718779   
8132  usa  United States  2015        382307000.0         319929162   
8133  usa  United States  2016        395881000.0         322179605   
8134  usa  United States  2017        395881000.0         324459463   

      avg_cell_pop  
8092      0.000000  
8093      0.000000  
8094      0.000000  
8095      0.000000  
8096      0.000000  
8097      0.000000  
8098      0.000000  
8099      0.000000  
8100      0.000000  
8101      0.000384  
8102      0.001413  
8103      0.002805  
8104      0.005016  
8105      0.008353  
8106      0.014029  
8107      0.020921  
8108      0.029639  
8109      0.042853  
8110      0.061570  
8111      0.091856  
8112      0.127177  
8113      0.163848  
8114      0.203252  
8115      0.251175  
8116      0.308565  
8117      0.388244  
8118      0.451111  
8119      0.493206  
8120      0.553868  
8121      0.631775  
8122      0.690205  
8123      0.770916  
8124      0.829355  
8125      0.861313  
8126      0.896126  
8127      0.923784  
8128      0.956125  
8129      0.972881  
8130      0.984665  
8131      1.118914  
8132      1.194974  

In [0]:
df2_cp = df1_cp.merge(df1_cp['avg_cell_pop'].to_frame())

In [24]:
df2_cp

geo      country  time  cell_phones_total  population_total  \
0         afg  Afghanistan  1960                0.0           8996351   
1         afg  Afghanistan  1960                0.0           8996351   
2         afg  Afghanistan  1960                0.0           8996351   
3         afg  Afghanistan  1960                0.0           8996351   
4         afg  Afghanistan  1960                0.0           8996351   
5         afg  Afghanistan  1960                0.0           8996351   
6         afg  Afghanistan  1960                0.0           8996351   
7         afg  Afghanistan  1960                0.0           8996351   
8         afg  Afghanistan  1960                0.0           8996351   
9         afg  Afghanistan  1960                0.0           8996351   
10        afg  Afghanistan  1960                0.0           8996351   
11        afg  Afghanistan  1960                0.0           8996351   
12        afg  Afghanistan  1960                0.0           8996351   
13        afg  Afghanistan  1960                0.0           8996351   
14        afg  Afghanistan  1960                0.0           8996351   
15        afg  Afghanistan  1960                0.0           8996351   
16        afg  Afghanistan  1960                0.0           8996351   
17        afg  Afghanistan  1960                0.0           8996351   
18        afg  Afghanistan  1960                0.0           8996351   
19        afg  Afghanistan  1960                0.0           8996351   
20        afg  Afghanistan  1960                0.0           8996351   
21        afg  Afghanistan  1960                0.0           8996351   
22        afg  Afghanistan  1960                0.0           8996351   
23        afg  Afghanistan  1960                0.0           8996351   
24        afg  Afghanistan  1960                0.0           8996351   
25        afg  Afghanistan  1960                0.0           8996351   
26        afg  Afghanistan  1960                0.0           8996351   
27        afg  Afghanistan  1960                0.0           8996351   
28        afg  Afghanistan  1960                0.0           8996351   
29        afg  Afghanistan  1960                0.0           8996351   
...       ...          ...   ...                ...               ...   
14104830  zmb       Zambia  2009          4406682.0          13456417   
14104831  zmb       Zambia  2010          5446991.0          13850033   
14104832  zmb       Zambia  2011          8164553.0          14264756   
14104833  zmb       Zambia  2012         10524676.0          14699937   
14104834  zmb       Zambia  2013         10395801.0          15153210   
14104835  zmb       Zambia  2014         10114867.0          15620974   
14104836  zmb       Zambia  2015         11557725.0          16100587   
14104837  zmb       Zambia  2016         12017034.0          16591390   
14104838  zmb       Zambia  2017         13438539.0          17094130   
14104839  zwe     Zimbabwe  1997             5734.0          11709997   
14104840  zwe     Zimbabwe  1998            19000.0          11893272   
14104841  zwe     Zimbabwe  1999           174000.0          12064537   
14104842  zwe     Zimbabwe  2000           266441.0          12222251   
14104843  zwe     Zimbabwe  2001           314002.0          12366165   
14104844  zwe     Zimbabwe  2002           338779.0          12500525   
14104845  zwe     Zimbabwe  2003           363651.0          12633897   
14104846  zwe     Zimbabwe  2004           425745.0          12777511   
14104847  zwe     Zimbabwe  2005           647110.0          12940032   
14104848  zwe     Zimbabwe  2006           849146.0          13124267   
14104849  zwe     Zimbabwe  2007          1225654.0          13329909   
14104850  zwe     Zimbabwe  2008          1654721.0          13558469   
14104851  zwe     Zimbabwe  2009          3991000.0          13810599   
14104852  zwe     Zimbabwe  2010          7700000.0          1408631

In [0]:
Avg_Cell_pop = 395881000/324459463

In [26]:
Avg_Cell_pop

1.2201246847283354

Modify the `geo` column to make the geo codes uppercase instead of lowercase.

In [0]:
df2_cp['geo'] =df2_cp['geo'].str.upper()

In [28]:
df2_cp.head()

geo      country  time  cell_phones_total  population_total  avg_cell_pop
0  AFG  Afghanistan  1960                0.0           8996351           0.0
1  AFG  Afghanistan  1960                0.0           8996351           0.0
2  AFG  Afghanistan  1960                0.0           8996351           0.0
3  AFG  Afghanistan  1960                0.0           8996351           0.0
4  AFG  Afghanistan  1960                0.0           8996351           0.0

## Part 3. Process data

Use the describe function, to describe your dataframe's numeric columns, and then its non-numeric columns.

(You'll see the time period ranges from 1960 to 2017, and there are 195 unique countries represented.)

In [29]:
df2_cp.describe()

time  cell_phones_total  population_total  avg_cell_pop
count  1.410486e+07       1.410486e+07      1.410486e+07  1.410486e+07
mean   1.981199e+03       5.484104e+03      2.076571e+07  1.703027e-04
std    8.957180e+00       1.393160e+06      8.609352e+07  1.316217e-02
min    1.960000e+03       0.000000e+00      4.433000e+03  0.000000e+00
25%    1.977000e+03       0.000000e+00      9.660310e+05  0.000000e+00
50%    1.982000e+03       0.000000e+00      4.371521e+06  0.000000e+00
75%    1.987000e+03       0.000000e+00      1.213307e+07  0.000000e+00
max    2.017000e+03       1.474097e+09      1.409517e+09  2.490243e+00

In [38]:
df2_cp.sort_values('time')

geo                         country  time  cell_phones_total  \
0         AFG                     Afghanistan  1960                0.0   
12669375  TTO             Trinidad and Tobago  1960                0.0   
12669374  TTO             Trinidad and Tobago  1960                0.0   
12669373  TTO             Trinidad and Tobago  1960                0.0   
12669372  TTO             Trinidad and Tobago  1960                0.0   
12669371  TTO             Trinidad and Tobago  1960                0.0   
12669370  TTO             Trinidad and Tobago  1960                0.0   
2816250   COM                         Comoros  1960                0.0   
2816251   COM                         Comoros  1960                0.0   
2816252   COM                         Comoros  1960                0.0   
2816253   COM                         Comoros  1960                0.0   
2816254   COM                         Comoros  1960                0.0   
12669376  TTO             Trinidad and Tobago  1960                0.0   
2816255   COM                         Comoros  1960                0.0   
2816257   COM                         Comoros  1960                0.0   
2816258   COM                         Comoros  1960                0.0   
2816259   COM                         Comoros  1960                0.0   
2816260   COM                         Comoros  1960                0.0   
2816261   COM                         Comoros  1960                0.0   
2816262   COM                         Comoros  1960                0.0   
2816263   COM                         Comoros  1960                0.0   
2816264   COM                         Comoros  1960                0.0   
2816265   COM                         Comoros  1960                0.0   
2816266   COM                         Comoros  1960                0.0   
2816267   COM                         Comoros  1960                0.0   
2816256   COM                         Comoros  1960                0.0   
12669377  TTO             Trinidad and Tobago  1960                0.0   
12669378  TTO             Trinidad and Tobago  1960                0.0   
12669379  TTO             Trinidad and Tobago  1960                0.0   
10731775  RUS                          Russia  1960                0.0   
...       ...                             ...   ...                ...   
14101197  DJI                        Djibouti  2017          373,052.0   
14104169  SWE                          Sweden  2017       12,435,709.0   
14100462  BGR                        Bulgaria  2017        8,532,908.0   
14102854  MDA                         Moldova  2017        3,662,968.0   
14101838  GTM                       Guatemala  2017       19,986,482.0   
14102620  LCA                       St. Lucia  2017          176,694.0   
14100619  BOL                         Bolivia  2017       10,963,224.0   
14102640  LIE                   Liechtenstein  2017           46,625.0   
14104663  VCT  St. Vincent and the Grenadines  2017          116,161.0   
14102668  LKA                       Sri Lanka  2017       28,199,083.0   
14104268  TGO                            Togo  2017        6,219,981.0   
14102690  LSO                         Lesotho  2017        2,380,804.0   
14101142  CZE                  Czech Republic  2017       12,634,937.0   
14101810  GRD                         Grenada  2017          113,177.0   
14100568  BLR                         Belarus  2017       11,415,141.0   
14100647  BRA                          Brazil  2017      236,488,548.0   
14102716  LTU                       Lithuania  2017        4,361,329.0   
14104230  SYR                           Syria  2017       15,650,000.0   
14101175  DEU                         Germany  2017      106,000,000.0   
14101782  GRC                          Greece  2017       12,937,106.0   
14102749  LUX                      Luxembourg  2017          794,000.0   
14102775  LVA                          Latvia  2017        2,464,122.0   
14100521  BHS 

In [56]:
df2_cp.sort_values('cell_phones_total', ascending=True)

geo        country  time  cell_phones_total  population_total  \
0         AFG    Afghanistan  1960                0.0           8996351   
9400011   NGA        Nigeria  1989                0.0          92844353   
9400012   NGA        Nigeria  1989                0.0          92844353   
9400013   NGA        Nigeria  1989                0.0          92844353   
9400014   NGA        Nigeria  1989                0.0          92844353   
9400015   NGA        Nigeria  1989                0.0          92844353   
9400016   NGA        Nigeria  1989                0.0          92844353   
9400017   NGA        Nigeria  1989                0.0          92844353   
9400018   NGA        Nigeria  1989                0.0          92844353   
9400019   NGA        Nigeria  1989                0.0          92844353   
9400020   NGA        Nigeria  1989                0.0          92844353   
9400010   NGA        Nigeria  1989                0.0          92844353   
9400021   NGA        Nigeria  1989                0.0          92844353   
9400023   NGA        Nigeria  1989                0.0          92844353   
9400024   NGA        Nigeria  1989                0.0          92844353   
9400025   NGA        Nigeria  1989                0.0          92844353   
9400026   NGA        Nigeria  1989                0.0          92844353   
9400027   NGA        Nigeria  1989                0.0          92844353   
9400028   NGA        Nigeria  1989                0.0          92844353   
9400029   NGA        Nigeria  1989                0.0          92844353   
9400030   NGA        Nigeria  1989                0.0          92844353   
9400031   NGA        Nigeria  1989                0.0          92844353   
9400032   NGA        Nigeria  1989                0.0          92844353   
9400022   NGA        Nigeria  1989                0.0          92844353   
9400009   NGA        Nigeria  1989                0.0          92844353   
9400008   NGA        Nigeria  1989                0.0          92844353   
9400007   NGA        Nigeria  1989                0.0          92844353   
9399984   NGA        Nigeria  1989                0.0          92844353   
9399985   NGA        Nigeria  1989                0.0          92844353   
9399986   NGA        Nigeria  1989                0.0          92844353   
...       ...            ...   ...                ...               ...   
14102028  IDN      Indonesia  2015      338,948,340.0         258162113   
14102044  IND          India  2008      346,890,000.0        1197146906   
14104609  USA  United States  2014      355,500,000.0         317718779   
14104610  USA  United States  2015      382,307,000.0         319929162   
14102029  IDN      Indonesia  2016      385,573,398.0         261115456   
14100871  CHN          China  2005      393,406,000.0        1321623490   
14104612  USA  United States  2017      395,881,000.0         324459463   
14104611  USA  United States  2016      395,881,000.0         322179605   
14102030  IDN      Indonesia  2017      458,923,202.0         263991379   
14100872  CHN          China  2006      461,058,000.0        1329209094   
14102045  IND          India  2009      525,090,000.0        1214270132   
14100873  CHN          China  2007      547,306,000.0        1336800506   
14100874  CHN          China  2008      641,245,000.0        1344415227   
14100875  CHN          China  2009      747,214,000.0        1352068091   
14102046  IND          India  2010      752,190,000.0        1230980691   
14100876  CHN          China  2010      859,003,000.0        1359755102   
14102048  IND          India  2012      864,720,917.0        1263065852   
14102049  IND          India  2013      886,304,245.0        1278562207   
14102047  IND          India  2011      893,862,478.0        1247236029   
14102050  IND          India  2014      944,008,677.0        1293859294   
14100877  CHN          China  2011      986,253,000.0        1367480264   
14102051  IND          India  2015 

In 2017, what were the top 5 countries with the most cell phones total?

Your list of countries should have these totals:

| country | cell phones total |
|:-------:|:-----------------:|
|   China    |     1,474,097,000 |
|    Inda    |     1,168,902,277 |
|   Indonesia    |       458,923,202 |
|    United States   |       395,881,000 |
|   Brazil    |       236,488,548 |



In [0]:
# This optional code formats float numbers with comma separators
pd.options.display.float_format = '{:,}'.format

2017 was the first year that China had more cell phones than people.

What was the first year that the USA had more cell phones than people?

2012

## Part 4. Reshape data

Create a pivot table:
- Columns: Years 2007—2017
- Rows: China, India, United States, Indonesia, Brazil (order doesn't matter)
- Values: Cell Phones Total

The table's shape should be: (5, 11)

In [71]:
df2_cp2 = df2_cp.pivot_table(index='country', columns='time',values='cell_phones_total')

print ('Pivot Table Shape:')
       
df2_cp2.shape

df2_cp2.head()

Pivot Table Shape:


time         1960  1965  1970  1975  1976  1977  1978  1979  1980  1981  \
country                                                                   
Afghanistan   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
Albania       0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
Algeria       0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
Andorra       0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
Angola        0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   

time            ...              2008         2009         2010         2011  \
country         ...                                                            
Afghanistan     ...       7,898,909.0 10,500,000.0 10,215,840.0 13,797,879.0   
Albania         ...       1,859,632.0  2,463,741.0  2,692,372.0  3,100,000.0   
Algeria         ...      27,031,472.0 32,729,824.0 32,780,165.0 35,615,926.0   
Andorra         ...          64,202.0     64,549.0     65,495.0     65,044.0   
Angola          ...       6,773,356.0  8,109,421.0  9,403,365.0 12,073,218.0   

time                2012         2013         2014         2015         2016  \
country                                                                        
Afghanistan 15,340,115.0 16,807,156.0 18,407,168.0 19,709,038.0 21,602,982.0   
Albania      3,500,000.0  3,685,983.0  3,359,654.0  3,400,955.0  3,369,756.0   
Algeria     37,527,703.0 39,517,045.0 43,298,174.0 43,227,643.0 47,041,321.0   
Andorra         63,865.0     63,931.0     66,241.0     71,336.0     76,132.0   
Angola      12,785,109.0 13,285,198.0 14,052,558.0 13,884,532.0 13,001,124.0   

time                2017  
country                   
Afghanistan 23,929,713.0  
Albania      3,497,950.0  
Algeria     49,873,389.0  
Andorra         80,337.0  
Angola      13,323,952.0  

[5 rows x 46 columns]

#### OPTIONAL BONUS QUESTION!

Sort these 5 countries, by biggest increase in cell phones from 2007 to 2017.

Which country had 935,282,277 more cell phones in 2017 versus 2007?

If you have the time and curiosity, what other questions can you ask and answer with this data?